applications/vanilla/models/class.discussionmodel.php

```diff
@@ -2985,6 +2985,24 @@ public function canView($discussion, $userID = 0) {
         return $canView;
     }
 
+    /**
+     * Tests whether a user has permission to view a specific discussion, with consideration for mod permissions.
+     *
+     * @param object $discussion
+     * @return bool whether the user can view a discussion.
+     */
+    public function canViewDiscussion($discussion): bool {
+        $canViewDiscussion = true;
```

Comment:
This is probably more of a style choice, but I do recommend you default to `false` and only flip it to `true` when you've confirmed the user has access to the discussion. If the default is `false`, we have to prove the user has that access.

Reply:
I agree. Defaulting to false it is.